### PR TITLE
fix(middleware/session): mutex for thread safety

### PR DESCRIPTION
**Pull request by sixcolors**
This pull request adds a mutex to the session middleware to ensure thread safety when accessing session data. The mutex is used to protect non-data fields and prevent race conditions. This improves the reliability and stability of the session middleware.

### CODE REVIEW

File: middleware/session/session_test.go [hunk 1]
```diff
@@ -673,3 +675,150 @@ func Benchmark_Session(b *testing.B) {
 		utils.AssertEqual(b, nil, err)
 	})
 }
+
+// go test -v -run=^$ -bench=Benchmark_Session_Asserted_Parallel -benchmem -count=4
+func Benchmark_Session_Asserted_Parallel(b *testing.B) {
```
Comment (gaby) [inline middleware/session/session_test.go:680]: Add another parallel benchmark without asserts

Comment (sixcolors) [reply middleware/session/session_test.go:680]: added all benchmark variations from v3.

**Comment by ReneWerner87** [comment]
@sixcolors i don't see any mutex adjustments?

**Comment by sixcolors** [comment]
> @sixcolors i don't see any mutex adjustments?

seems I medded up the merge, give me a couple minutes to fix.

**Comment by sixcolors** [comment]
@ReneWerner87 ready

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**